### PR TITLE
Support multilingual names in instance, fix fallback order

### PR DESCRIPTION
**Pull request by NightFurySL2001**

### CODE REVIEW

**Comment by anthrotype** [comment]
thanks. Would be nice to also add a test

**Comment by NightFurySL2001** [comment]
@anthrotype tests added that showed a few more edge cases.

File: Lib/ufo2ft/instantiator.py [hunk 1]
```diff
@@ -596,40 +689,141 @@ def _generate_instance_info(
...
+                        instance.localisedStyleMapFamilyName[language_tag],
+                        language_tag,
+                    )
+                )
+            elif language_tag in instance.localisedFamilyName:
+                # Style Map Family Name (ID=1)
+                # autofallback to ID=16 + styleName ID=17 (but omit ID=17 if it's standard style)
+                full_family_name = instance.localisedFamilyName[language_tag]
+                if language_tag in instance.localisedStyleName:
+                    # if the styleMapStyleName is set, use it
```
Comment (anthrotype) [inline Lib/ufo2ft/instantiator.py:791]: this inline comment doesn't make sense to me, it's referring to `styleMapStyleName` but you're handling `localisedStyleName` here..

**Comment by anthrotype** [comment]
@NightFurySL2001 one test is still failing, see https://github.com/googlefonts/ufo2ft/actions/runs/16247324710/job/45988814800?pr=921#step:5:148

**Review by anthrotype — APPROVED** [review]

**Comment by anthrotype** [comment]
@NightFurySL2001 Thanks. Would you like to squash commits before I merge?

**Comment by NightFurySL2001** [comment]
Can you directly squash and merge? Might be more preferable than force pushing to the repo (i think?).

**Comment by NightFurySL2001** [comment]
Squashed and ready
